Extend/Controller/OrderController.php

```diff
@@ -273,10 +369,126 @@ private function _isPaymentFinished($oSessionToken, $oPaymentRedirect)
      * Checks if the returned order creation state is one of the success states
      *
      * @param int $iSuccess
+     *
      * @return bool
      */
     private function _isFinalizeOrderSuccessful($iSuccess)
     {
         return $iSuccess === Order::ORDER_STATE_MAILINGERROR || $iSuccess === Order::ORDER_STATE_OK;
     }
+
+    /**
+     *
+     * Returns the parameters used to render the credit card form
+     *
+     * @return string
+     * @throws \Exception
+     */
+    public function getInitCreditCardFormJavaScript(): string
+    {
+
+        /**
+         * @var $oPaymentGateway PaymentGateway
+         */
+        $oPaymentGateway = oxNew(PaymentGateway::class);
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+
+        /**
+         * @var $oOrder Order
+         */
+        $oOrder = oxNew(Order::class);
+        $oOrder->createTemp($oBasket, $this->getUser());
+
+        $oTransaction = $oPaymentGateway->createTransaction($oBasket, $oOrder);
+
+        Registry::getLogger()->debug("getInitCreditCardFormJavaScript Transaction:" . print_r($oTransaction, true));
+
+        $oTransaction->setAmount(new Amount(
+            $oBasket->getPrice()->getBruttoPrice(),
+            $oBasket->getBasketCurrency()->name
+        ));
+        $oTransaction->setConfig($this->_getPaymentMethodConfig()->get(CreditCardTransaction::NAME));
+
+        $oSession = $this->getSession();
+        $sSid = $oSession->sid(true);
+        if ($sSid != '') {
+            $sSid = '&' . $sSid;
```

Comment:
You can also use `http_build_query` for this.

Reply:
:+1: